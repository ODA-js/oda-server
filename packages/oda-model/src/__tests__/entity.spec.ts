import 'jest';
// import { Entity } from '..';
// import { AsHash } from '../types';
// import { FieldInput } from '../field';
// import { SimpleFieldInput } from '../simplefield';

describe('entity', () => {
  it('loads', () => {
    // const res = new Entity({
    //   name: 'A',
    //   fields: {
    //     name: {
    //       type: 'Number',
    //     } as Partial<SimpleFieldInput>,
    //   } as AsHash<FieldInput>,
    // });
    // expect(res).toMatchSnapshot('raw');
    // expect(res.toObject()).toMatchSnapshot('toObject');
  });
  it('id', () => {
    // const res = new Entity({
    //   name: 'A',
    //   fields: {
    //     id: {} as Partial<SimpleFieldInput>,
    //     name: {
    //       type: 'Number',
    //     } as Partial<SimpleFieldInput>,
    //   } as AsHash<FieldInput>,
    // });
    // expect(res).toMatchSnapshot('id-raw');
    // expect(res.toObject()).toMatchSnapshot('id-toObject');
  });
  it('_id', () => {
    // const res = new Entity({
    //   name: 'A',
    //   fields: {
    //     _id: {} as Partial<SimpleFieldInput>,
    //     name: {
    //       type: 'Number',
    //     } as Partial<SimpleFieldInput>,
    //   } as AsHash<FieldInput>,
    // });
    // expect(res).toMatchSnapshot('_id=raw');
    // expect(res.toObject()).toMatchSnapshot('_id-toObject');
  });
});
