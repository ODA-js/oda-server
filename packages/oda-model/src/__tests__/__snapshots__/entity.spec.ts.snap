// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entity _id: _id=raw 1`] = `
Entity {
  "$obj": Object {
    "abstract": false,
    "embedded": false,
    "fields": Map {
      "_id" => SimpleField {
        "$obj": Object {
          "args": Map {},
          "idKey": EntityReference {
            "$obj": Object {
              "backField": "id",
              "entity": "A",
              "field": "_id",
              Symbol(metadata): Object {},
            },
          },
          "list": false,
          "name": "_id",
          Symbol(metadata): Object {
            "acl": Object {
              "read": Array [],
              "update": Array [],
            },
            "entity": "A",
            "order": 0,
            "persistence": Object {
              "identity": true,
              "indexed": true,
              "required": true,
            },
          },
        },
      },
      "name" => SimpleField {
        "$obj": Object {
          "args": Map {},
          "list": false,
          "name": "name",
          "type": "Number",
          Symbol(metadata): Object {
            "acl": Object {
              "read": Array [],
              "update": Array [],
            },
            "entity": "A",
            "order": 1,
            "persistence": Object {},
          },
        },
      },
    },
    "identity": Set {
      "_id",
    },
    "implements": Set {},
    "indexed": Set {
      "_id",
    },
    "name": "A",
    "operations": Map {},
    "relations": Set {},
    "required": Set {
      "_id",
    },
    Symbol(metadata): Object {
      "UI": Object {
        "edit": Array [],
        "embedded": Array [],
        "hidden": Array [],
        "list": Array [],
        "listName": Array [],
        "quickSearch": Array [],
        "show": Array [],
      },
      "acl": Object {
        "create": Array [],
        "delete": Array [],
        "read": Array [],
        "upadte": Array [],
      },
      "name": Object {
        "plural": "As",
        "singular": "A",
      },
      "persistence": Object {
        "indexes": Object {
          "_id": Object {
            "fields": Object {
              "_id": 1,
            },
            "name": "_id",
            "options": Object {
              "unique": true,
            },
          },
        },
      },
      "titlePlural": "As",
    },
  },
}
`;

exports[`entity _id: _id-toObject 1`] = `
Object {
  "abstract": false,
  "description": undefined,
  "embedded": false,
  "fields": Array [
    Object {
      "args": Object {},
      "defaultValue": undefined,
      "derived": false,
      "description": undefined,
      "entity": "A",
      "identity": true,
      "indexed": true,
      "inheritedFrom": undefined,
      "list": false,
      "metadata": Object {
        "acl": Object {
          "read": Array [],
          "update": Array [],
        },
        "entity": "A",
        "order": 0,
        "persistence": Object {
          "identity": true,
          "indexed": true,
          "required": true,
        },
      },
      "name": "_id",
      "order": 0,
      "persistent": false,
      "required": true,
      "title": undefined,
      "type": undefined,
    },
    Object {
      "args": Object {},
      "defaultValue": undefined,
      "derived": false,
      "description": undefined,
      "entity": "A",
      "identity": false,
      "indexed": false,
      "inheritedFrom": undefined,
      "list": false,
      "metadata": Object {
        "acl": Object {
          "read": Array [],
          "update": Array [],
        },
        "entity": "A",
        "order": 1,
        "persistence": Object {},
      },
      "name": "name",
      "order": 1,
      "persistent": false,
      "required": false,
      "title": undefined,
      "type": "Number",
    },
  ],
  "implements": Array [],
  "metadata": Object {
    "UI": Object {
      "edit": Array [],
      "embedded": Array [],
      "hidden": Array [],
      "list": Array [],
      "listName": Array [],
      "quickSearch": Array [],
      "show": Array [],
    },
    "acl": Object {
      "create": Array [],
      "delete": Array [],
      "read": Array [],
      "upadte": Array [],
    },
    "name": Object {
      "plural": "As",
      "singular": "A",
    },
    "persistence": Object {
      "indexes": Object {
        "_id": Object {
          "fields": Object {
            "_id": 1,
          },
          "name": "_id",
          "options": Object {
            "unique": true,
          },
        },
      },
    },
    "titlePlural": "As",
  },
  "name": "A",
  "operations": Array [],
  "title": undefined,
}
`;

exports[`entity id: id-raw 1`] = `
Entity {
  "$obj": Object {
    "abstract": false,
    "embedded": false,
    "fields": Map {
      "id" => SimpleField {
        "$obj": Object {
          "args": Map {},
          "idKey": EntityReference {
            "$obj": Object {
              "backField": "id",
              "entity": "A",
              "field": "id",
              Symbol(metadata): Object {},
            },
          },
          "list": false,
          "name": "id",
          Symbol(metadata): Object {
            "acl": Object {
              "read": Array [],
              "update": Array [],
            },
            "entity": "A",
            "order": 0,
            "persistence": Object {
              "identity": true,
              "indexed": true,
              "required": true,
            },
          },
        },
      },
      "name" => SimpleField {
        "$obj": Object {
          "args": Map {},
          "list": false,
          "name": "name",
          "type": "Number",
          Symbol(metadata): Object {
            "acl": Object {
              "read": Array [],
              "update": Array [],
            },
            "entity": "A",
            "order": 1,
            "persistence": Object {},
          },
        },
      },
    },
    "identity": Set {
      "id",
    },
    "implements": Set {},
    "indexed": Set {
      "id",
    },
    "name": "A",
    "operations": Map {},
    "relations": Set {},
    "required": Set {
      "id",
    },
    Symbol(metadata): Object {
      "UI": Object {
        "edit": Array [],
        "embedded": Array [],
        "hidden": Array [],
        "list": Array [],
        "listName": Array [],
        "quickSearch": Array [],
        "show": Array [],
      },
      "acl": Object {
        "create": Array [],
        "delete": Array [],
        "read": Array [],
        "upadte": Array [],
      },
      "name": Object {
        "plural": "As",
        "singular": "A",
      },
      "persistence": Object {
        "indexes": Object {
          "id": Object {
            "fields": Object {
              "id": 1,
            },
            "name": "id",
            "options": Object {
              "unique": true,
            },
          },
        },
      },
      "titlePlural": "As",
    },
  },
}
`;

exports[`entity id: id-toObject 1`] = `
Object {
  "abstract": false,
  "description": undefined,
  "embedded": false,
  "fields": Array [
    Object {
      "args": Object {},
      "defaultValue": undefined,
      "derived": false,
      "description": undefined,
      "entity": "A",
      "identity": true,
      "indexed": true,
      "inheritedFrom": undefined,
      "list": false,
      "metadata": Object {
        "acl": Object {
          "read": Array [],
          "update": Array [],
        },
        "entity": "A",
        "order": 0,
        "persistence": Object {
          "identity": true,
          "indexed": true,
          "required": true,
        },
      },
      "name": "id",
      "order": 0,
      "persistent": false,
      "required": true,
      "title": undefined,
      "type": undefined,
    },
    Object {
      "args": Object {},
      "defaultValue": undefined,
      "derived": false,
      "description": undefined,
      "entity": "A",
      "identity": false,
      "indexed": false,
      "inheritedFrom": undefined,
      "list": false,
      "metadata": Object {
        "acl": Object {
          "read": Array [],
          "update": Array [],
        },
        "entity": "A",
        "order": 1,
        "persistence": Object {},
      },
      "name": "name",
      "order": 1,
      "persistent": false,
      "required": false,
      "title": undefined,
      "type": "Number",
    },
  ],
  "implements": Array [],
  "metadata": Object {
    "UI": Object {
      "edit": Array [],
      "embedded": Array [],
      "hidden": Array [],
      "list": Array [],
      "listName": Array [],
      "quickSearch": Array [],
      "show": Array [],
    },
    "acl": Object {
      "create": Array [],
      "delete": Array [],
      "read": Array [],
      "upadte": Array [],
    },
    "name": Object {
      "plural": "As",
      "singular": "A",
    },
    "persistence": Object {
      "indexes": Object {
        "id": Object {
          "fields": Object {
            "id": 1,
          },
          "name": "id",
          "options": Object {
            "unique": true,
          },
        },
      },
    },
    "titlePlural": "As",
  },
  "name": "A",
  "operations": Array [],
  "title": undefined,
}
`;

exports[`entity loads: raw 1`] = `
Entity {
  "$obj": Object {
    "abstract": false,
    "embedded": false,
    "fields": Map {
      "name" => SimpleField {
        "$obj": Object {
          "args": Map {},
          "list": false,
          "name": "name",
          "type": "Number",
          Symbol(metadata): Object {
            "acl": Object {
              "read": Array [],
              "update": Array [],
            },
            "entity": "A",
            "order": 0,
            "persistence": Object {},
          },
        },
      },
      "id" => SimpleField {
        "$obj": Object {
          "args": Map {},
          "idKey": EntityReference {
            "$obj": Object {
              "backField": "id",
              "entity": "a",
              "field": "id",
              Symbol(metadata): Object {},
            },
          },
          "list": false,
          "name": "id",
          "type": "ID",
          Symbol(metadata): Object {
            "acl": Object {
              "read": Array [],
              "update": Array [],
            },
            "entity": "a",
            "persistence": Object {
              "identity": true,
              "indexed": true,
              "required": true,
            },
          },
        },
      },
    },
    "identity": Set {
      "id",
    },
    "implements": Set {},
    "indexed": Set {
      "id",
    },
    "name": "A",
    "operations": Map {},
    "relations": Set {},
    "required": Set {
      "id",
    },
    Symbol(metadata): Object {
      "UI": Object {
        "edit": Array [],
        "embedded": Array [],
        "hidden": Array [],
        "list": Array [],
        "listName": Array [],
        "quickSearch": Array [],
        "show": Array [],
      },
      "acl": Object {
        "create": Array [],
        "delete": Array [],
        "read": Array [],
        "upadte": Array [],
      },
      "name": Object {
        "plural": "As",
        "singular": "A",
      },
      "persistence": Object {
        "indexes": Object {
          "id": Object {
            "fields": Object {
              "id": 1,
            },
            "name": "id",
            "options": Object {
              "unique": true,
            },
          },
        },
      },
      "titlePlural": "As",
    },
  },
}
`;

exports[`entity loads: toObject 1`] = `
Object {
  "abstract": false,
  "description": undefined,
  "embedded": false,
  "fields": Array [
    Object {
      "args": Object {},
      "defaultValue": undefined,
      "derived": false,
      "description": undefined,
      "entity": "A",
      "identity": false,
      "indexed": false,
      "inheritedFrom": undefined,
      "list": false,
      "metadata": Object {
        "acl": Object {
          "read": Array [],
          "update": Array [],
        },
        "entity": "A",
        "order": 0,
        "persistence": Object {},
      },
      "name": "name",
      "order": 0,
      "persistent": false,
      "required": false,
      "title": undefined,
      "type": "Number",
    },
    Object {
      "args": Object {},
      "defaultValue": undefined,
      "derived": false,
      "description": undefined,
      "entity": "a",
      "identity": true,
      "indexed": true,
      "inheritedFrom": undefined,
      "list": false,
      "metadata": Object {
        "acl": Object {
          "read": Array [],
          "update": Array [],
        },
        "entity": "a",
        "persistence": Object {
          "identity": true,
          "indexed": true,
          "required": true,
        },
      },
      "name": "id",
      "order": undefined,
      "persistent": false,
      "required": true,
      "title": undefined,
      "type": "ID",
    },
  ],
  "implements": Array [],
  "metadata": Object {
    "UI": Object {
      "edit": Array [],
      "embedded": Array [],
      "hidden": Array [],
      "list": Array [],
      "listName": Array [],
      "quickSearch": Array [],
      "show": Array [],
    },
    "acl": Object {
      "create": Array [],
      "delete": Array [],
      "read": Array [],
      "upadte": Array [],
    },
    "name": Object {
      "plural": "As",
      "singular": "A",
    },
    "persistence": Object {
      "indexes": Object {
        "id": Object {
          "fields": Object {
            "id": 1,
          },
          "name": "id",
          "options": Object {
            "unique": true,
          },
        },
      },
    },
    "titlePlural": "As",
  },
  "name": "A",
  "operations": Array [],
  "title": undefined,
}
`;
